crates/nostr-sdk/src/client/mod.rs

```diff
@@ -1101,17 +1101,56 @@ impl Client {
         self.send_event_to(urls, event).await
     }
 
-    /// Get public key metadata
+    /// Fetch public key metadata from database, falling back to connected relays.
```

Comment:
This is not a real fallback. This will always query the relays. But could be an idea to query relays only if the database query return an empty result. Could be a new option in the `EventSource` like `database_with_fallback`. This will be useful too for gossip.

Reply:
That's why such documentation is essential, thanks for the clarification ;) but then why involve the database at all?
In which case would the database contain something that the relays don't?

Reply:
Yeah, I know, current documentation sucks.

Since nostr relays aren't like bitcoin nodes could always happen. It's possible that a set of connected relays not have a specific event while the database yes because received it in the past from another relay (which could be offline now).